core/src/main/java/google/registry/model/billing/BillingEvent.java

```diff
@@ -347,27 +347,11 @@ public DateTime getSyntheticCreationTime() {
 
     @Override
     public VKey<OneTime> createVKey() {
-      return VKey.create(
-          getClass(),
-          parent.getParent().getName() + "/" + parent.getId() + "/" + getId(),
-          Key.create(this));
+      return VKey.create(getClass(), getId(), Key.create(this));
     }
 
     public static VKey<OneTime> createVKey(Key<OneTime> key) {
-      // TODO(b/159207551): As it stands, the SQL key generated here doesn't mesh with the primary
-      // key type for the table, which is a single long integer.
-      if (key == null) {
-        return null;
-      }
-      Key parent = key.getParent();
-      Key grandparent = (parent != null) ? parent.getParent() : null;
```

Comment:
It seems like this code was trying to solve some kind of problem that will now not be solved by the code on the right side of this diff?  What is the long term plan here?

Reply:
Or will this be sufficient once the duplicate IDs are taken care of?

Reply:
1. The old code was not trying to solve a problem, instead, it aimed to make any caller fail if they try to use the SQL key from the VKey. As it mentioned above, the SQL table for PollMessage and BillingEvent use a long type column as its primary key but the old code constructed a string for it. Ideally, this should work because none of the production code uses the SQL key from the VKey. However, the process to construct the SQL key has a bug in PollMessage(not the code you highlighted), i.e. `key.getParent().getParent()` would throw a NPE when `key` doesn't have its grandparent.
2. I assume you were asking if the new code, i.e. `key == null ? null : VKey.create(OneTime.class, key.getId(), key);` will be sufficient and the answer is yes, because our plan is to use the numeric id of the Ofy key as the primary key for its SQL table.